webapp/shop/cube/views.py

```diff
@@ -710,3 +713,88 @@ def cred_syllabus_data(**kawrgs):
         syllabus_data=syllabus_data,
         exam_name=exam_name,
     )
+
+
+@shop_decorator(area="cube", permission="user", response="html")
+def cred_shop(**kwargs):
+    return flask.render_template("credentials/shop/index.html")
+
+
+@shop_decorator(area="cube", permission="user", response="html")
+def cred_redeem_code(ua_contracts_api, advantage_mapper, **kwargs):
```

Comment:
http://localhost:8001/credentials/redeem is visible when not logged in and always displays a success notification. This should be behind SSO and show an error if no activation key was provided.